test/core/security/credentials_test.cc

```diff
@@ -3701,6 +3693,72 @@ TEST(CredentialsTest, TestXdsCredentialsCompareFailure) {
   grpc_channel_credentials_release(xds_creds_2);
 }
 
+TEST(TlsCertificateVerifierTest, ComparingDifferentObjectTypesFails) {
```

Comment:
The verifier tests belong in test/core/security/grpc_tls_certificate_verifier_test.cc.